### ISSUE TITLE
Infinite/bad loop in ESCAPE DETECTOR

### ISSUE THREAD

**Issue by smartguy1196** [smartguy1196/regex.objectified#31]
the loop checks to see if another match exists in the string, does not account for whether the STRING TRACKER VALUE went up. If it does not go up, it will just check again.

**Comment by smartguy1196** [comment]
This needs to be added to issue #26, and mentioned in issue #22

**Comment by smartguy1196** [comment]
issue is reopened, because it was not added to #26

**Comment by smartguy1196** [comment]
Actually this issue is no longer relevant to issue #26, but needs to be added/mentioned in #22

**Comment by smartguy1196** [comment]
cycler_sync.detector should only increment if on the last token in the cycle

**Comment by smartguy1196** [comment]
Fix (added to #22):

```
( debug.cycler_sync.detector = //parenthesis: assign a value to debug.cycler_sync.detector, then return (that) debug.cycler_sync.detector's value
  ( loop.st_i === debug.string_tracker.length - 1 ? if the STRING TRACKER INDEX (index of STRING TRACKER) is the last index
    ( debug.detector_sync.prev_dt_i === loop.dt_i ? //and if the STRING TRACKER VALUE (index of token) did not change
      /*incremented value: (in ESCAPE DETECTOR) 1*/ : //increment the value
      /*kept/maintained/static value: (in ESCAPE DETECTOR) 0*/ //otherwise maintain the value at the same state
    ) :
    /*kept/maintained/static value: (in ESCAPE DETECTOR) 0*/ //otherwise maintain the value at the same state
  )
) === /*value associated with detector: (in ESCAPE DETECTOR) 0*/
```
